app/stac_api/utils.py

```diff
@@ -444,3 +455,27 @@ def get_s3_cache_control_value(update_interval):
         return f'max-age={max_age}, public'
 
     return f'max-age={settings.STORAGE_ASSETS_CACHE_SECONDS}, public'
+
+
+def select_s3_bucket(collection_name) -> AVAILABLE_S3_BUCKETS:
+    """Select the s3 bucket based on the collection name
+
+    Select the correct s3 bucket based on matching patterns with the collection
+    name
+    """
+    patterns = settings.MANAGED_BUCKET_COLLECTION_PATTERNS
+
+    for pattern in patterns:
+        match = re.fullmatch(pattern, collection_name)
+        if match is not None:
+            return AVAILABLE_S3_BUCKETS.MANAGED
+
+    return AVAILABLE_S3_BUCKETS.LEGACY
+
+
+def get_aws_settings(s3_bucket: AVAILABLE_S3_BUCKETS = AVAILABLE_S3_BUCKETS.LEGACY):
+
+    if s3_bucket == AVAILABLE_S3_BUCKETS.LEGACY:
+        return settings.AWS_LEGACY
+
+    return settings.AWS_MANAGED
```

Comment:
we could also just use a dict with bucket-name as keys in the settings, then we wouldn't need an additional function. what do you think?

Reply:
Yes good idea. Will change that. This would also be more elegant in the future when we potentially remove the old bucket.